Server/ConcordiaCurriculumManager/Controllers/GroupController.cs

```diff
@@ -79,16 +107,76 @@ public async Task<IActionResult> AddUserToGroup(Guid groupId, Guid userId)
     }
 
     [HttpDelete("{groupId}/users/{userId}")]
-    [Authorize(Roles = RoleNames.Admin)]
+    [Authorize]
     [SwaggerResponse(StatusCodes.Status200OK, "User removed from group successfully")]
     [SwaggerResponse(StatusCodes.Status404NotFound, "Error removing user from group")]
     public async Task<IActionResult> RemoveUserFromGroup(Guid groupId, Guid userId)
     {
+        Guid currentUserId;
+        try
+        {
+            currentUserId = Guid.Parse(_userService.GetCurrentUserClaim(Claims.Id));
+        }
+        catch (Exception e)
+        {
+            var message = "Unexpected error occured while retrieving user information.";
+            _logger.LogWarning($"${message}: {e.Message}");
+            return Problem(
+                title: message,
+                detail: e.Message,
+                statusCode: StatusCodes.Status500InternalServerError);
+        }
+
+        bool isAdmin = User.IsInRole(RoleNames.Admin);
+        bool isMaster = await _groupService.IsGroupMaster(currentUserId, groupId);
+
+        if (!isAdmin && !isMaster)
+        {
+            return StatusCode(StatusCodes.Status403Forbidden, "You are not authorized to add users to this group");
+        }
+
         var result = await _groupService.RemoveUserFromGroup(userId, groupId);
         if (result)
         {
             return Ok();
         }
         return NotFound();
     }
+
+    [HttpPost("{groupId}/masters/{userId}")]
+    [Authorize]
+    [SwaggerResponse(StatusCodes.Status200OK, "User added to group masters successfully")]
+    [SwaggerResponse(StatusCodes.Status404NotFound, "User is not a member of this group or is already a group master")]
+    public async Task<IActionResult> AddGroupMaster(Guid groupId, Guid userId)
```

Comment:
In addition to AddGroupMaster, can you also do the RemoveGroupMaster endpoint? It should be very similar altogether.